File: automation/troubleshooting/ARCHIVED/start_console.js
const { spawn } = require('child_process');
const path = require('path');

// Environment variables
process.env.FEATURE_MESSAGE_BROKER_TYPE = 'nats';
process.env.NATS_ADDRESS = '127.0.0.1';
process.env.NATS_PORT = '8222';
process.env.NSQ_HOST = '127.0.0.1';
process.env.NSQ_PORT = '8222';

// Working directory
const workingDir = '/var/opt/kaspersky/ksc-web-console';
process.chdir(workingDir);

// Command: ./node pm.js ./pm.config.js
const cmd = './node';
const args = ['pm.js', './pm.config.js'];

console.log(`Starting KSC Web Console from ${workingDir}...`);
console.log(`Command: ${cmd} ${args.join(' ')}`);

const child = spawn(cmd, args, {
  cwd: workingDir,
  env: process.env,
  stdio: 'inherit'
});

child.on('close', (code) => {
  console.log(`KSC Web Console exited with code ${code}`);
  process.exit(code);
});

child.on('error', (err) => {
  console.error('Failed to start child process.', err);
  process.exit(1);
});

File: automation/troubleshooting/ARCHIVED/start_console.py
import subprocess
import os
import sys


def start_console():
    # Environment variables
    env = os.environ.copy()
    env["FEATURE_MESSAGE_BROKER_TYPE"] = "nats"
    env["NATS_ADDRESS"] = "127.0.0.1"
    env["NATS_PORT"] = "4222"
    env["NSQ_HOST"] = "127.0.0.1"
    env["NSQ_PORT"] = "4222"

    # Working directory
    working_dir = "/var/opt/kaspersky/ksc-web-console"
    os.chdir(working_dir)

    # Command
    # We use ./node as it is the bundled version
    cmd = ["./node", "pm.js", "./pm.config.js"]

    print(f"Starting KSC Web Console from {working_dir}...")
    print(f"Command: {' '.join(cmd)}")

    try:
        # Run and stream output
        process = subprocess.Popen(
            cmd, env=env, stdout=sys.stdout, stderr=sys.stderr, universal_newlines=True
        )
        process.wait()
    except Exception as e:
        print(f"Error starting console: {e}")
        sys.exit(1)


if __name__ == "__main__":
    start_console()
